addons/sale/views/sale_views.xml

```diff
@@ -439,7 +555,14 @@
                     <field name="user_id"/>
                     <field name="team_id" string="Sales Channel"/>
                     <field name="analytic_account_id"/>
-                    <field name="product_id"/>
+                    <!-- We only allow to search on the following sale order line fields (product, name) because the other fields, such as price, quantity, ...
+                        will not be searched as often, and if they need to be searched it's usually in the context of products
+                        and then they can be searched from the page listing the sale order lines related to a product (from the product itself).
+                    -->
+                    <field name="order_line" string="Line Product" filter_domain="[('order_line.product_id.name', 'ilike', self)]"/>
```

Comment:
Isn't it an issue having several times the same field name ? Moreover what happens if someone makes an xpath based on name ?

Reply:
There was no technical issue with having the same field name in a search.

I'll however remove the others for now because they make the view more complicated for little gain.

If a client needs it, a consultant can easily add it for them.